de.dlr.sc.overtarget.language.tests/src/de/dlr/sc/overtarget/language/generator/util/ReferencedTargetHelperTest.xtend

```diff
@@ -177,11 +173,54 @@ class ReferencedTargetHelperTest {
 			}".getBytes();
 		val sourceTarget2 = new ByteArrayInputStream(bytes2);
 		tmodelFile2.create(sourceTarget2, IResource.NONE, null);
-		
-		Resource.Factory.Registry.INSTANCE.getExtensionToFactoryMap().put("tmodel", resourceFactory);
-		val testTmodelResource2 = rs.getResource(URI.createPlatformResourceURI("/testProject2/target/target2.tmodel", true), true)
-		val tmodel2 = testTmodelResource2.contents.get(0) as TargetModel
-		
-		Assert.assertNull(refTargetHelper.findTargetfileOfTmodel(tmodel2, outputDirectory, uri))
+
+		Assert.assertNull(refTargetHelper.findTargetFileInProject(tmodelFile2, outputDirectory))
+	}
+	
+	@Test
+	def void testFindTargetForReferencesFile() {
+		val outputDirectory = "./target"
+		val workspace = ResourcesPlugin.getWorkspace();
+		val root = workspace.getRoot();
+		val project  = root.getProject("testProject3");
+		val folder = project.getFolder("target");
+		val tmodelFile = folder.getFile("targetForReferences.tmodel");
+		project.create(null);
+		if (!project.isOpen()) { 
+			project.open(null)
+		}
+		folder.create(IResource.NONE, true, null);
+
+		val bytes = "
+			Target targetForReferences {
+				
+			}".getBytes();
+		val source = new ByteArrayInputStream(bytes);
+		tmodelFile.create(source, IResource.NONE, null);
+		val targetFile = folder.getFile("targetForReferences.target");
+
+		val bytesTarget = 
+			''''''.toString.getBytes();
+		val sourceTarget = new ByteArrayInputStream(bytesTarget)
+		targetFile.create(sourceTarget, false, null)
+
+		Assert.assertEquals(targetFile, refTargetHelper.findTargetForReferencesFile(tmodelFile, outputDirectory))
+
+		val project2 = root.getProject("testProject4");
+		val folder2 = project2.getFolder("target");
+		val tmodelFile2 = folder2.getFile("targetForReferences.tmodel");
+		project2.create(null);
+		if (!project2.isOpen()) { 
+			project2.open(null)
+		}
+		folder2.create(IResource.NONE, true, null);
+		val bytes2 = "
+			Target targetForReferences {
+				
+			}".getBytes();
+		val sourceTarget2 = new ByteArrayInputStream(bytes2);
+		tmodelFile2.create(sourceTarget2, IResource.NONE, null);
+
+		Assert.assertNull(refTargetHelper.findTargetForReferencesFile(tmodelFile2, outputDirectory))
```

Comment:
Why do we expect NULL here? Maybe add a message in the assert statement.... 
`Assert.assertNull("File does not exist so shoud be null", refTargetHelper.findTargetForReferencesFile(tmodelFile2, outputDirectory))`